lib/streamlit/elements/widgets/button_group.py

```diff
@@ -403,61 +403,55 @@ def _pills(
         kwargs: WidgetKwargs | None = None,
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
-    ) -> list[V] | V | None:
-        maybe_raise_label_warnings(label, label_visibility)
-
-        def _transformed_format_func(option: V) -> ButtonGroupProto.Option:
-            """If option starts with a material icon or an emoji, we extract it to send
-            it parsed to the frontend."""
-            transformed = format_func(option) if format_func else str(option)
-            transformed_parts = transformed.split(" ")
-            icon: str | None = None
-            if len(transformed_parts) > 0:
-                maybe_icon = transformed_parts[0].strip()
-                try:
-                    # we only want to extract material icons because we treat them
-                    # differently than emojis visually
-                    if maybe_icon.startswith(":material"):
-                        icon = validate_material_icon(maybe_icon)
-                        # reassamble the option string without the icon - also
-                        # works if len(transformed_parts) == 1
-                        transformed = " ".join(transformed_parts[1:])
-                except StreamlitAPIException:
-                    # we don't have a valid icon or emoji, so we just pass
-                    pass
-            return ButtonGroupProto.Option(
-                content=transformed,
-                content_icon=icon,
-            )
-
-        indexable_options = convert_to_sequence_and_check_comparable(options)
-        default_values = get_default_indices(indexable_options, default)
-
-        serde: SingleOrMultiSelectSerde[V] = SingleOrMultiSelectSerde[V](
-            indexable_options, default_values, selection_mode
-        )
-        res = self._button_group(
-            indexable_options,
-            key=key,
-            default=default_values,
+    ):
```

Comment:
nit: missing return type. All our functions should be typed, but especially the ones that are publicly exposed (this one has just been made visible again in https://github.com/streamlit/streamlit/pull/9612). Same for the `def segmented_control` function below.